src/classes/Countertop.js

```diff
@@ -40,13 +48,69 @@ class Countertop {
 			},
 		)
 		this.stations.push(station)
+		this.updateTopology()
 		return station
 	}
 
+	/**
+	 * Start the countertop and begin data processing.
+	 *
+	 * This will regenerate the countertop topology to reflect all currently added appliances.
+	 *
+	 * @return {Boolean} Whether the countertop successfully started.
+	 */
+	start = async () => {
+		this.logger.trace('CountertopCoordinator: start()')
+		this.state = countertopStates.STARTING
+		const promises = this.stations.map(async (station) => station.start())
+		if ((await Promise.all(promises)).every((result) => result)) {
+			this.state = countertopStates.STARTED
+		} else {
+			this.state = countertopStates.ERRORED
+		}
+		return this.getState() === countertopStates.STARTED
+	}
+
+	/**
+	 * Stop the countertop and halt all data processing.
+	 *
+	 * @return {Boolean} Whether the countertop successfully stopped.
+	 */
+	stop = async () => {
+		this.logger.trace('CountertopCoordinator: stop()')
+		this.state = countertopStates.STOPPING
+		this.stations.map((station) => station.stop())
+		const promises = this.stations.map(async (station) => station.stop())
+		if ((await Promise.all(promises)).every((result) => result)) {
```

Comment:
I wonder if instead of jamming the promise one-liner into this if statement, this wouldn't be cleaner as a variable.